src/client/Microsoft.Identity.Client/Instance/Discovery/InstanceDiscoveryManager.cs

```diff
@@ -109,14 +108,26 @@ internal class InstanceDiscoveryManager : IInstanceDiscoveryManager
             RequestContext requestContext, 
             bool forceValidation = false)
         {
-
             Uri authorityUri = authorityInfo.CanonicalAuthority;
             string environment = authorityInfo.Host;
 
             if (authorityInfo.IsInstanceDiscoverySupported)
             {
                 var entry = _userMetadataProvider?.GetMetadataOrThrow(environment, requestContext.Logger);
```

Comment:
I would throw if app dev sets their own metadata and this flag at the same time.

Reply:
I think I was incorrect in assuming that user metadata was a separate concept from network metadata. So, in actuality, usermatada should also be ignored when IsInstanceDiscoverySupported is set to false. Which is why @bgavrilMS is suggesting to throw if both are set.

Reply:
Are we talking about IsInstanceDiscoverySupported  or InstanceDiscoveryEndpoint?
I think it's valid if user provides metadata and disables network discovery. In this case we just check user + regional + cached + known metadta.

Reply:
Im talking about IsInstanceDiscoverySupported. Also, not sure about "if user provides metadata and disables network discovery. In this case we just check user + regional + cached + known metadta."

Reply:
Yeah. I think the current implemenation around user metadata is also fine taking precedence over this flag is also fine.